cg/store/crud/read.py

```diff
@@ -1545,3 +1542,61 @@ def get_case_ids_for_samples(self, sample_ids: list[int]) -> list[str]:
         for sample_id in sample_ids:
             case_ids.extend(self.get_case_ids_with_sample(sample_id))
         return list(set(case_ids))
+
+    def get_related_dna_samples_from_sample_within_collaborators(
+        self, sample_internal_id: str, collaborators: set[Customer]
+    ) -> list[Sample]:
+        """Returns a DNA sample with the same subject_id, tumour status and within the collaborators of an RNA sample."""
+        sample: Sample = self.get_sample_by_internal_id(internal_id=sample_internal_id)
+
+        sample_application_version_query: Query = self._get_join_sample_application_version_query()
+        dna_prep_categories: list[PrepCategory] = [
+            PrepCategory.WHOLE_GENOME_SEQUENCING,
+            PrepCategory.TARGETED_GENOME_SEQUENCING,
+            PrepCategory.WHOLE_EXOME_SEQUENCING,
+        ]
+        sample_application_version_query: Query = apply_application_filter(
+            applications=sample_application_version_query,
+            prep_categories=dna_prep_categories,
+            filter_functions=[ApplicationFilter.BY_PREP_CATEGORIES],
+        )
+
+        customer_entry_ids: list[str] = [customer.id for customer in collaborators]
+
+        sample_application_version_query: Query = apply_sample_filter(
+            samples=sample_application_version_query,
+            subject_id=sample.subject_id,
+            is_tumour=sample.is_tumour,
+            customer_entry_ids=customer_entry_ids,
+            filter_functions=[
+                SampleFilter.BY_SUBJECT_ID,
+                SampleFilter.BY_TUMOUR,
+                SampleFilter.BY_CUSTOMER_ENTRY_IDS,
+            ],
+        )
+
+        return sample_application_version_query.all()
+
+    def get_cases_from_sample_by_workflows_within_collaborators(
+        self, sample_internal_id: str, workflows: list[Workflow], collaborators: set[Customer]
+    ) -> list[Case]:
+        """Return a list of DNA cases linked to DNA sample with a list of customers in a collaboration."""
+
+        cases_with_samples: Query = self._join_sample_and_case()
+        cases_with_samples: Query = apply_case_sample_filter(
+            case_samples=cases_with_samples,
+            sample_internal_id=sample_internal_id,
+            filter_functions=[CaseSampleFilter.CASES_WITH_SAMPLE_BY_INTERNAL_ID],
+        )
+
+        customer_entry_ids: list[str] = [customer.id for customer in collaborators]
```

Comment:
Can the list comprehension be moved to the function call below instead of using an extra variable?

Reply:
Yes!